components/config-mgmt-service/backend/elastic/runs.go

```diff
@@ -71,12 +72,142 @@ func (es Backend) GetRun(runID string, endTime time.Time) (backend.Run, error) {
 	return run, nil
 }
 
+// GetDeletedCountsTimeSeries - creates a daily time series of the total amount of nodes that have
+// been deleted within or before the time range. The time series is between the startTime and endTime provided.
+// The startTime and endTime must atleast be 24 hours apart. When greater than 24 hours, it must be in
+// multiples 24 hour blocks.
+// A node is deleted if exists = false. The time the node was deleted is the "timestamp"
+func (es Backend) GetDeletedCountsTimeSeries(
+	startTime, endTime time.Time, filters map[string][]string) ([]backend.CountPeroid, error) {
+	var (
+		aggTag = "date_range"
+	)
+	copiedFilters := map[string][]string{}
+	for index, element := range filters {
+		copiedFilters[index] = element
+	}
+	copiedFilters["exists"] = []string{"false"}
+	mainQuery := newBoolQueryFromFilters(copiedFilters)
+
+	deletedNodePeroids := make([]backend.CountPeroid, getNumberOf24hBetween(startTime, endTime))
+
+	dateRangeAgg := elastic.NewDateRangeAggregation().Field(backend.Timestamp).
```

Comment:
This is a type of aggregation I have not used before. The "Field" is the date to create the buckets from. Below you can see we are creating custom buckets. Each bucket is collecting all the nodes that have been deleted before the time range. 
https://www.elastic.co/guide/en/elasticsearch/reference/6.8/search-aggregations-bucket-daterange-aggregation.html

Reply:
nicely done, there's a lot of work in here!